carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,260 @@
+        Constructing the ML model
+
+        Parameters
+        ----------
+        model_type : str
+            Name of the model ``{name}.{ext}`` on https://github.com/indyfree/cf-models.
+        data_name : str
+            Name of the dataset the model has been trained on.
+        cache : boolean, optional
+            If True, try to load from the local cache first, and save to the cache
+            if a download is required.
+        models_home : string, optional
+            The directory in which to cache data; see :func:`get_models_home`.
+        kws : keys and values, optional
+            Additional keyword arguments are passed to passed through to the read model function
+        data : data.api.Data Class
+            Correct dataset for ML model
+        ext : String
+            File extension of saved ML model file
+        """
+        # check if dataset fits to ML model
+        assert data.name == data_name
+
+        self._continuous = data.continous
+        self._categoricals = data.categoricals
+
+        self._data_name = data_name
+        self._model = load_model(model_type, data_name, ext, cache, models_home, **kws)
+
+        if ext == "pt":
+            self._backend = "pytorch"
+        elif ext == "h5":
+            self._backend = "tensorflow"
+        else:
+            raise Exception("Model type not in catalog")
+
+        self._name = model_type + "_" + data_name
+
+        # Preparing pipeline components
+        self._scaler = preprocessing.MinMaxScaler().fit(data.raw[self._continuous])
+
+        if data_name == "adult":
+            self._feature_input_order = [
+                "age",
+                "fnlwgt",
+                "education-num",
+                "capital-gain",
+                "capital-loss",
+                "hours-per-week",
+                "workclass_Private",
+                "marital-status_Non-Married",
+                "occupation_Other",
+                "relationship_Non-Husband",
+                "race_White",
+                "sex_Male",
+                "native-country_US",
+            ]
+            self._encodings = (
+                [  # Encodings should be built in the get_dummy way: {column}_{value}
+                    "workclass_Private",
+                    "marital-status_Non-Married",
+                    "occupation_Other",
+                    "relationship_Non-Husband",
+                    "race_White",
+                    "sex_Male",
+                    "native-country_US",
+                ]
+            )
+        else:
+            raise Exception("Model for dataset not in catalog")
+
+    def pipeline(self, df):
+        """
+        Transforms input for prediction into correct form.
+        Only possible for DataFrames without preprocessing steps.
+
+        Recommended to use to keep correct encodings, normalization and input order
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            Contains unnormalized and
```

Comment:
Seems like docs are missing here

Reply:
Yes that's strange :smile: Will add them :+1: